c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -845,17 +862,28 @@ public:
   kj::Promise<void> connect(ConnectContext context) override {
     auto params = context.getParams();
     auto host = params.getHost();
-    kj::HttpConnectSettings settings = {
-      .useTls = params.getSettings().getUseTls()
-    };
+    kj::Own<kj::TlsStarterCallback> tlsStarter = kj::heap<kj::TlsStarterCallback>();
+    kj::HttpConnectSettings settings = { .useTls = params.getSettings().getUseTls()};
+    settings.tlsStarter = tlsStarter;
     auto headers = factory.headersToKj(params.getHeaders());
     auto pipe = kj::newTwoWayPipe();
 
     class EofDetector final: public kj::AsyncOutputStream {
     public:
-      EofDetector(kj::Own<kj::AsyncIoStream> inner) : inner(kj::mv(inner)) {}
+      EofDetector(kj::Own<kj::AsyncIoStream> inner, kj::TlsStarterCallback& starter)
+          : inner(kj::mv(inner)), starter(starter) {}
       ~EofDetector() {
         inner->shutdownWrite();
+        // Revoke the TLS starter when stream is destroyed. This will ensure any startTls calls
```

Comment:
Could we instead pass ownership of the `TlsStarterCallback` directly to the `ByteStream` and let it take care of dropping it in end()?

Then we don't need to allocate `TlsStarterCallback` as a heap object and pass all these references around.